### PR TITLE
Rename Authorize to affirm

### PR DESCRIPTION
**Pull request by satyamakgec**
- [x] TODO runtime upgrade.

### CODE REVIEW

File: pallets/settlement/src/lib.rs [hunk 1]
```diff
@@ -540,85 +536,79 @@ decl_module! {
...
         ///
         /// # Arguments
         /// * `venue_id` - ID of the venue this instruction belongs to.
         /// * `settlement_type` - Defines if the instruction should be settled
-        ///    immediately after receiving all auths or waiting till a specific block.
+        ///    immediately after receiving all affirms or waiting till a specific block.
         /// * `valid_from` - Optional date from which people can interact with this instruction.
         /// * `legs` - Legs included in this instruction.
-        /// * `portfolios` - Portfolios that the sender controls and wants to authorize for this instruction
+        /// * `portfolios` - Portfolios that the sender controls and wants to affirm this instruction
```
Comment (maxsam4) [inline pallets/settlement/src/lib.rs:552]: This doesn't sound right. Maybe try something like
```suggestion
        /// * `portfolios` - Portfolios that the sender controls and wants to use in this affirmation
```

File: pallets/settlement/src/lib.rs [hunk 2]
```diff
@@ -1184,24 +1167,22 @@ impl<T: Trait> Module<T> {
...
         instruction_id: u64,
         portfolios: BTreeSet<PortfolioId>,
     ) -> DispatchResult {
         Self::ensure_instruction_validity(instruction_id)?;
-
-        // checks portfolio's custodian and if it is a counter party with a pending or rejected authorization
+        // checks portfolio's custodian and if it is a counter party with a pending or rejected affirmation
         for portfolio in &portfolios {
-            let user_auth = Self::user_auths(portfolio, instruction_id);
+            let user_auth = Self::user_affirms(portfolio, instruction_id);
```
Comment (maxsam4) [inline pallets/settlement/src/lib.rs:1182]:
```suggestion
            let user_affirmation = Self::user_affirms(portfolio, instruction_id);
```

**Review by adamdossa — APPROVED** [review]

File: pallets/settlement/src/lib.rs [hunk 3]
```diff
@@ -22,8 +22,8 @@
 //!
 //! - `create_venue` - Registers a new venue.
 //! - `add_instruction` - Adds a new instruction.
-//! - `authorize_instruction` - Authorizes an existing instruction.
-//! - `unauthorize_instruction` - Unauthorizes an existing instruction.
+//! - `affirm_instruction` - Provides affirmation to an existing instruction.
+//! - `withdraw_instruction` - Deny an existing instruction.
```
Comment (maxsam4) [inline pallets/settlement/src/lib.rs:26]: `withdraw_instruction` is not correct. The function withdraws the `affirmation`, not the instruction.

File: pallets/settlement/src/lib.rs [hunk 4]
```diff
@@ -340,10 +340,10 @@ decl_event!(
...
             Vec<Leg<Balance>>,
         ),
-        /// An instruction has been authorized (did, portfolio, instruction_id)
-        InstructionAuthorized(IdentityId, PortfolioId, u64),
-        /// An instruction has been unauthorized (did, portfolio, instruction_id)
-        InstructionUnauthorized(IdentityId, PortfolioId, u64),
+        /// An instruction has been affirmed (did, portfolio, instruction_id)
+        InstructionAffirmed(IdentityId, PortfolioId, u64),
+        /// An instruction has been denied (did, portfolio, instruction_id)
+        InstructionDenied(IdentityId, PortfolioId, u64),
```
Comment (maxsam4) [inline pallets/settlement/src/lib.rs:346]: `InstructionDenied` -> `AffirmationWithdrawn`

**Review by maxsam4 — CHANGES_REQUESTED** [review]

**Review by maxsam4 — APPROVED** [review]

**Pull request merged**